### PR TITLE
Create Tabs pixar component

### PR DESCRIPTION
**Pull request by grosbouddha**
A basic Tabs component made of:
   - a list of selectable tabs (only one can be selected at a time)
   - one panel: its content is controlled by the selected tab

It follows most of the ARIA authoring recommendations for a Tabs component from:
https://www.w3.org/TR/wai-aria-practices-1.1/#tabpanel
In particular, it implements most of the recommended shortcuts, role structure and is using
a roving tabindex strategy for the focus management.

Screenshot:
image

Important note: For this component, only the API and a11y is final. The visual and motion specs/implementation might change in a future iteration.

### CODE REVIEW

File: src/tabs/style.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  border-bottom: 2px solid ${color.white};
+  color: ${color.primaryText};
+  font-size: ${font.base.size};
+  white-space: nowrap;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  flex: 1;
+  flex-grow: 1;
+  flex-shrink: 1;
+  flex-basis: 0%;
```
Comment (Tuxosaurus) [inline src/tabs/style.ts]: nit: 0 without unit

File: src/tabs/index.unit.tsx [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    expect(tabpanel.text()).toBe('Content for second tab')
+
+    // Simulate a click on tab 1.
+    wrapper.find('[role="tab"]').at(0).simulate('click')
+
+    tabpanel = wrapper.find('[role="tabpanel"]')
+    expect(tabpanel.text()).toBe('Content for first tab')
+})
+
+it('Should left/right keyboard navigate', () => {
```
Comment (pith) [inline src/tabs/index.unit.tsx]: nit: you could use a `describe` for your global description then split this test with multiple `it`s

**Comment by Tuxosaurus** [comment]
Not sure if it's because of the integration within Storybook but navigating with Tabs should first select a tab and then the corresponding panel. I get a "white screen" when tabbing to the panel.

File: src/tabs/index.tsx [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+/**
+ * A basic Tabs component made of:
+ *  - a list of selectable tabs (only one can be selected at a time)
+ *  - one panel: its content is controlled by the selected tab
+ *
+ * It follows most of the ARIA authoring recommendations for a Tabs component from:
+ * https://www.w3.org/TR/wai-aria-practices-1.1/#tabpanel
+ * In particular, it implements most of the recommended shortcuts, role structure and is using
+ * a roving tabindex for the focus management.
+ */
```
Comment (Tip-Sy) [inline src/tabs/index.tsx]: 👍 :notbad: 😉

File: src/tabs/index.unit.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+    // Verify aria-selected
+    expect(tabs.get(0).props['aria-selected']).toBe('true')
+    expect(tabs.get(1).props['aria-selected']).toBe('false')
+    expect(tabs.get(2).props['aria-selected']).toBe('false')
+    // Verify tabIndexes
+    expect(tabs.get(0).props['tabIndex']).toBe(0)
+    expect(tabs.get(1).props['tabIndex']).toBe(-1)
+    expect(tabs.get(2).props['tabIndex']).toBe(-1)
+
+    // verify that the current tabpanel id is correct.
```
Comment (Tip-Sy) [inline src/tabs/index.unit.tsx]: nit: capital V 😛

File: src/tabs/index.unit.tsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    // Verify aria-selected
+    expect(tabs.get(0).props['aria-selected']).toBe('false')
+    expect(tabs.get(1).props['aria-selected']).toBe('true')
+    expect(tabs.get(2).props['aria-selected']).toBe('false')
+    // Verify tabIndexes
+    expect(tabs.get(0).props['tabIndex']).toBe(-1)
+    expect(tabs.get(1).props['tabIndex']).toBe(0)
+    expect(tabs.get(2).props['tabIndex']).toBe(-1)
+
+    // verify that the current tabpanel id is correct.
```
Comment (Tip-Sy) [inline src/tabs/index.unit.tsx]: nit: capital V 😛

File: src/tabs/style.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+
+export default css`
+.kirk-tablist {
+  display: flex;
+}
+
+.kirk-tab {
+  outline: none;
+  text-transform: uppercase;
+  padding: 8px 16px 6px 16px;
```
[the remark this answers is not in the document]
Comment (grosbouddha) [reply src/tabs/style.ts]: Done. thanks Nico!

File: src/tabs/index.unit.tsx (hunk 2 above)
Comment (grosbouddha) [reply src/tabs/index.unit.tsx]: Done, thanks!

File: src/tabs/style.ts (hunk 1 above)
Comment (grosbouddha) [reply src/tabs/style.ts]: Done

File: src/tabs/index.unit.tsx (hunk 5 above)
Comment (grosbouddha) [reply src/tabs/index.unit.tsx]: Done

**Comment by grosbouddha** [comment]
> Not sure if it's because of the integration within Storybook but navigating with the tab key should first select a tab and then the corresponding panel. I get a "white screen" when tabbing to the panel.

Good catch @Tuxosaurus ! I forgot to check for the TAB case in my keydown handler and it was exploding. I'm now checking explicitly and I added a test to cover that.

**Pull request merged**